### PR TITLE
[route53_health_check] Consider port in identity test

### PR DESCRIPTION
**Pull request by mjfroehlich**
##### SUMMARY
Consider port when testing whether a health check already exists.

Before, asking for 2 health checks being present for `http://myhost:8888` and `http://myhost:9999`  resulted in just a single health check, as the module saw the two specs as equal.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
route53_health_check

##### ANSIBLE VERSION
```
ansible 2.4.0
  config file =
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Mar  4 2017, 14:56:39) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
```

### CODE REVIEW

**Review by s-hertel — APPROVED** [review]

**Comment by s-hertel** [comment]
shipit

File: lib/ansible/modules/cloud/amazon/route53_health_check.py [hunk 1]
```diff
@@ -156,7 +156,7 @@ def find_health_check(conn, wanted):
     """Searches for health checks that have the exact same set of immutable values"""
     for check in conn.get_list_health_checks().HealthChecks:
         config = check.HealthCheckConfig
-        if config.get('IPAddress') == wanted.ip_addr and config.get('FullyQualifiedDomainName') == wanted.fqdn and config.get('Type') == wanted.hc_type and config.get('RequestInterval') == str(wanted.request_interval):
+        if config.get('IPAddress') == wanted.ip_addr and config.get('FullyQualifiedDomainName') == wanted.fqdn and config.get('Type') == wanted.hc_type and config.get('RequestInterval') == str(wanted.request_interval) and config.get('Port') == str(wanted.port):
```
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/route53_health_check.py]: The logic is fine but this line is ridiculously long (it was before, now it's even longer)

Can you put the if condition in brackets and add a line break or two (the brackets mean you don't need an escaping `\` at the end of the line.)

**Comment by mjfroehlich** [comment]
@willthames happy you mention that! The coding guidelines left me with the impression that the minimal change is preferred ("we don't really care about line lengths.  Buy a bigger monitor!",  "no line continuations"), but I agree it's nicer this way - done in https://github.com/ansible/ansible/pull/22762/commits/44b0a98563430e0c9fc125b45afdbcdbe16bc71d

**Comment by mjfroehlich** [comment]
Thanks @ryansb !

**Review by ryansb — APPROVED** [review]

**Comment by willthames** [comment]
@mjfroehlich I think the general rule is to not change too much stuff just for the sake of passing standards, but certainly if you're in the general area, it's worth fixing things up. And given that there are now people driving closer adherence to standards, I tend to do a quick tidy as a separate commit if it's not too huge a job.

The current line length standard (as per tox.ini) is 160 - I tend to favour around 120 if I'm tidying up.

Thanks for fixing it, and glad to see this merged!

**Pull request merged**